pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -286,89 +289,129 @@ func (n *NetworkPolicyController) processClusterNetworkPolicy(cnp *crdv1alpha1.C
 				// For ACNPs that have per-namespace rules, cluster-level rules will be created with appliedTo
 				// set as the spec appliedTo for each rule.
-				if appliedToPerRule && len(cnp.Spec.AppliedTo) > 0 {
-					ruleAppliedTos = cnp.Spec.AppliedTo
+				if appliedToPerRule && len(acnp.Spec.AppliedTo) > 0 {
+					ruleAppliedTos = acnp.Spec.AppliedTo
 				}
 				ruleATGNames := n.processClusterAppliedTo(ruleAppliedTos, atgNamesSet)
-				klog.V(4).Infof("Adding a new cluster-level rule with appliedTos %v for %s", ruleATGNames, cnp.Name)
-				addRule(n.toAntreaPeerForCRD(clusterPeers, cnp, direction, namedPortExists), direction, ruleATGNames)
+				klog.V(4).InfoS("Adding a new cluster-level rule with appliedTos a ClusterNetworkPolicy", "appliedToGroupNames", ruleATGNames, "clusterNetworkPolicy", acnp.Name)
+				addRule(n.toAntreaPeerForCRD(clusterPeers, acnp, direction, namedPortExists), services, direction, ruleATGNames)
 			}
 			if len(perNSPeers) > 0 {
-				if len(cnp.Spec.AppliedTo) > 0 {
+				if len(acnp.Spec.AppliedTo) > 0 {
 					// Create a rule for each affected Namespace of appliedTo at spec level
 					for i := range clusterAppliedToAffectedNS {
-						klog.V(4).Infof("Adding a new per-namespace rule with appliedTo %v for rule %d of %s", clusterAppliedToAffectedNS[i], idx, cnp.Name)
-						addRule(n.toNamespacedPeerForCRD(perNSPeers, clusterAppliedToAffectedNS[i]), direction, []string{atgForNamespace[i]})
+						klog.V(4).InfoS("Adding a new per-namespace rule with appliedTo a ClusterNetworkPolicy rule", "appliedToAffectedNS", clusterAppliedToAffectedNS[i], "ruleIndex", idx, "clusterNetworkPolicy", acnp.Name)
+						addRule(n.toNamespacedPeerForCRD(perNSPeers, clusterAppliedToAffectedNS[i]), services, direction, []string{atgForNamespace[i]})
 					}
 				} else {
 					// Create a rule for each affected Namespace of appliedTo at rule level
-					for _, at := range cnpRule.AppliedTo {
+					for _, at := range acnpRule.AppliedTo {
 						affectedNS, selectors := n.getAffectedNamespacesForAppliedTo(at)
 						affectedNamespaceSelectors = append(affectedNamespaceSelectors, selectors...)
 						for _, ns := range affectedNS {
 							atg := n.createAppliedToGroup(ns, at.PodSelector, nil, at.ExternalEntitySelector)
 							atgNamesSet.Insert(atg)
-							klog.V(4).Infof("Adding a new per-namespace rule with appliedTo %v for rule %d of %s", atg, idx, cnp.Name)
-							addRule(n.toNamespacedPeerForCRD(perNSPeers, ns), direction, []string{atg})
+							klog.V(4).InfoS("Adding a new per-namespace rule with appliedTo for a ClusterNetworkPolicy rule", "appliedToGroup", atg, "ruleIndex", idx, "clusterNetworkPolicy", acnp.Name)
+							addRule(n.toNamespacedPeerForCRD(perNSPeers, ns), services, direction, []string{atg})
 						}
 					}
 				}
 			}
+			klog.InfoS("Checking cluster or perNSPeers", "numberOfClusterPeers", len(clusterPeers), "numberOfPerNamespacePeers", len(perNSPeers))
+		} else {
+			// Handle toService rules.
```

Comment:
what would be a good place to log some kind of warning if the ServiceType is not ClusterIP?

Reply:
Maybe we could add it inside `toAntreaServicesAndPeersFromServiceReference`. Previously, I used targetPort of Service as antreaService then I realized that I could just use the port in Endpoints. So I removed the part using serviceLister to get the Service. 
If we want this warning log, we need to get the Service to check its ServiceType. Not sure if it is worth doing this just for log?

Reply:
I realized that we can't just use Endpoints. Because for a non-selector Service, when we delete the Service, the Endpoints will still be there. And we should realize any rules for a non-existent Service. So no matter what we need to check if the Service exists. We could add ServiceType log after that check.
I have committed the check and log code. You can take a look. @antoninbas